libbeat/autodiscover/providers/kubernetes/pod.go

```diff
@@ -334,19 +334,17 @@ func (p *pod) emitEvents(pod *kubernetes.Pod, flag string, containers []kubernet
 		// This must be an id that doesn't depend on the state of the container
 		// so it works also on `stop` if containers have been already deleted.
 		eventID := fmt.Sprintf("%s.%s", pod.GetObjectMeta().GetUID(), c.Name)
-
-		cmeta := common.MapStr{
-			"id":      cid,
-			"name":    c.Name,
-			"image":   c.Image,
-			"runtime": runtimes[c.Name],
-		}
-		meta := p.metagen.Generate(pod, metadata.WithFields("container.name", c.Name),
-			metadata.WithFields("container.image", c.Image))
+		
+		meta := p.metagen.Generate(
+			pod,
+			metadata.WithFields("container.name", c.Name),
+			metadata.WithFields("container.image", c.Image),
+			metadata.WithFields("container.id", cid),
+			metadata.WithFields("container.runtime", runtimes[c.Name]),
```

Comment:
I think `kubernetes.container.id` and `kubernetes.container.runtime` don't have a mapping, could you confirm and add it?

I think kubernetes fields are defined in `libbeat/processors/add_kubernetes_metadata/_meta/fields.yml`.

Reply:
added